terraform example to configure entra id for entra id integration

flyinghermit opened the pull request.

flyinghermit committed:
terraform module to configure entra id for teleport entra id integration

flyinghermit committed:
terraform fmt

flyinghermit committed:
update README

r0mant reviewed (approved):
lgtm assuming this works OOTB

smallinsky reviewed (commented):
It looks like the current Terraform setup is missing the Entra ID  provider configuration.

For our IaC examples, it would be helpful if everything were ready to run end-to-end setup otherwise the flow becomes a bit cumbersome.


Ideally, I  should be able to do:
```
$ git clone teleport/exmaple
$ cd teleport/examples/terraform/entra-id-integration
// flow the README: 


cat > variables.auto.tfvars << EOF
app_name                = "teleport-entraid-integration"
proxy_service_address   = "example.teleport.sh"
certificate_expiry_date = "2026-05-01T01:02:03Z" # Warning! An expired certificate will break user authentication.

use_system_credentials   = true
graph_permission_ids    = [<permission IDs>]
managed_id              = "<principal ID of the managed identity>"
EOF



$ terraform init
$ terraform plan
// got clientId tenentID and past this in CLI entra ID plugin commend
```

However, without the provider block I’m seeing:
```
$ terraform plan

Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: Invalid provider configuration
│
│ Provider "registry.terraform.io/hashicorp/azuread" requires explicit configuration. Add a provider block to the root module and configure the provider's required arguments as described in the provider
│ documentation.
│
╵
╷
│ Error: unable to build authorizer: could not configure AzureCli Authorizer: could not parse Azure CLI version: launching Azure CLI: exec: "az": executable file not found in $PATH
│
│   with provider["registry.terraform.io/hashicorp/azuread"],
│   on <empty> line 0:
│   (source code not available)
```


As reference you can check: https://github.com/gravitational/teleport/blob/master/examples/athena/athena.tf e2e setup flow 


Where at last step it output the part of the teleprot.yaml: 

https://github.com/gravitational/teleport/blob/master/examples/athena/athena.tf#L269

flyinghermit committed:
review suggestions:
- add azure provider block and ask for tenant id input
- add client id and tenant id output
- update readme to set up azure cli

flyinghermit commented:
https://github.com/gravitational/teleport/pull/58252/commits/f9e44e17961a082bd5aafa3e2b84c2e6c2e9848b
- Added the `azuread` provider block along with `tenant_id` var. 
- Readme is also updated to set up Azure CLI. With this and the newly added provider block with `tenant_id` should at least reduce the provider configuration step. 
- Client ID and tenant ID output.

flyinghermit commented on examples/terraform/entra-id-integration/README.md:
```diff
@@ -0,0 +1,68 @@
+## Teleport Entra ID integration
```
yeah the plan was to followup with a link once the guide PR merges. I have added a tentative url now.
